### PR TITLE
Add TstampNanoSecs to block header

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
Current dependencies on/for this PR:
* feature/pos-consensus-skeleton
  * **PR #628** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/628" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/628?utm_source=stack-comment).

File: lib/block_view_bitcoin_test.go [hunk 1]
```diff
@@ -57,10 +58,7 @@ func GetTestParamsCopy(
 		// blocks that follow it, it hurdles the min chain work.
 		big.NewInt(0),
 		// We are bastardizing the DeSo header to store Bitcoin information here.
-		&MsgDeSoHeader{
-			TstampSecs: uint64(startHeader.Timestamp.Unix()),
-			Height:     0,
-		},
+		(&MsgDeSoHeader{Height: 0}).SetTstampSecs(uint64(startHeader.Timestamp.Unix())),
```
Comment (lazynina) [inline lib/block_view_bitcoin_test.go:61]: I don't love the way this looks. We could just set `TstampNanoSecs` on the struct, but I see why we did it this way to make the change less significant.

File: lib/postgres.go [hunk 2]
```diff
@@ -1288,9 +1289,11 @@ func (postgres *Postgres) UpsertBlockTx(tx *pg.Tx, blockNode *BlockNode) error {
 
 		TxMerkleRoot: blockNode.Header.TransactionMerkleRoot,
 		Version:      blockNode.Header.Version,
-		Timestamp:    blockNode.Header.TstampSecs,
-		Nonce:        blockNode.Header.Nonce,
-		ExtraNonce:   blockNode.Header.ExtraNonce,
+		// TODO: Does it matter that the postgres block timestamps are now in nanoseconds?
+		// Postgres in its current form should be deprecated by the time this matters.
```
[the remark this answers is not in the document]
Comment (tholonious) [reply lib/postgres.go]: Done. I've added a postgres migration to do the 1e9 scaling

**Review by lazynina — APPROVED** [review]

**Comment by tholonious** [comment]
@tholonious started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/628).

**Comment by tholonious** [comment]
@tholonious merged this pull request with [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/628).

**Pull request merged**